### PR TITLE
Use binary search instead of sequential search of sorted array in CDF function. Rename variable x to domain to carry more precious meaning.

### PR DESCRIPTION
**Pull request by chenshengoogle**
Please take a look.

### CODE REVIEW

File: stat/stat_test.go [hunk 1]
```diff
@@ -1374,65 +1377,22 @@ func TestCDF(t *testing.T) {
...
+					}
+					if v != test.ans[k][j] && !(math.IsNaN(v) && math.IsNaN(test.ans[k][j])) {
+						t.Errorf("mismatch case %d kind %d percentile %v. Expected: %v, found: %v", i, k, q, test.ans[k][j], v)
+					}
+				})
 			}
 		}
 	}
 
-	// these test cases should all result in a panic
```
Comment (kortschak) [inline stat/stat_test.go:1391]: Why are these removed?

Comment (chenshengoogle) [reply stat/stat_test.go:1391]: Sorry. This was mistakenly removed. I will fix it in the pull request.

**Comment by chenshengoogle** [comment]
> Can you provide benchmarks results that show this is worth the complexity?

Sure. Will add the benchmarks results in the pull request.

**Comment by vladimir-ch** [comment]
@chenshengoogle Do you plan to add the benchmarks? Otherwise I'll close this PR. Thanks.

**Comment by chenshengoogle** [comment]
Please close this PR. Thanks
